src/api/app/controllers/webui/users/tokens_controller.rb

```diff
@@ -105,19 +106,28 @@ def set_package
     # Check if only project_name or only package_name are present
     if @extra_params[:project_name].present? ^ @extra_params[:package_name].present?
       flash[:error] = 'When providing an optional package, both Project name and Package name must be provided.'
-      redirect_to new_token_path and return
+      @token = reinitialize_token(@params)
```

Comment:
You can move this `@token` assignment and the one below in just one line above the line 105